packages/side-nav/src/SideNavGroup/SideNavGroup.tsx

```diff
@@ -121,152 +109,48 @@ function SideNavGroup({
-
   // generate shared props for collapsible and static headers
   const groupHeaderProps = {
     'data-testid': 'side-nav-group-header-label',
     id: menuGroupLabelId,
   };
 
+  const sharedProps = {
+    groupHeaderProps,
+    indentLevel,
+    renderHeader: () => renderHeader(),
+    menuGroupLabelId,
+  };
+
   return (
     <li className={cx(listItemStyle, className)} {...rest}>
       {collapsible ? (
+        // collapsed, all items are inside a dropdown
         <>
-          <button
-            {...groupHeaderProps}
-            aria-controls={menuId}
-            aria-expanded={open}
-            className={cx(
-              buttonClassName,
-              headerStyle,
-              collapsibleHeaderStyle,
-              css`
-                width: ${width}px;
-              `,
-              {
-                [collapsibleHeaderFocusStyle]: usingKeyboard,
-                [indentedStyle]: indentLevel > 1,
-              },
-            )}
-            onClick={() => setOpen(curr => !curr)}
-          >
-            {renderedHeader}
-            <ChevronRight
-              role="presentation"
-              size={12}
-              className={cx(expandIconStyle, {
-                [openExpandIconStyle]: open,
-              })}
-            />
-          </button>
-
-          <Transition
-            in={open}
-            appear
-            timeout={150}
-            nodeRef={nodeRef}
-            mountOnEnter
-            unmountOnExit
-          >
-            {(state: TransitionStatus) => (
-              <div
-                ref={nodeRef}
-                className={cx(
-                  sideNavCollapsibleGroupBaseStyles,
-                  transitionStyles[state],
-                )}
-              >
-                <ul
-                  ref={ulRef}
-                  id={menuId}
-                  aria-labelledby={menuGroupLabelId}
-                  className={cx(
-                    ulStyleOverrides,
-                    css`
-                      transition: opacity ${transitionDuration.default}ms
-                        ease-in-out;
-                      opacity: 0;
-                    `,
-                    {
-                      [css`
-                        opacity: 1;
-                      `]: ['entering', 'entered'].includes(state),
-                    },
-                  )}
-                >
-                  {renderedChildren}
-                </ul>
-              </div>
-            )}
-          </Transition>
-        </>
-      ) : (
-        // not collapsible
-        <>
-          <div
-            {...groupHeaderProps}
-            className={cx(headerStyle, {
-              [indentedStyle]: indentLevel > 1,
-            })}
-          >
-            {renderedHeader}
-          </div>
-
-          <ul aria-labelledby={menuGroupLabelId} className={ulStyleOverrides}>
+          <SideNavGroupCollapsed open={open} setOpen={setOpen} {...sharedProps}>
             {renderedChildren}
```

Comment:
This feels like a subcomponent too. Although it's up to you whether you want to touch this since it doesn't really affect darkMode styles

Reply:
which one feels like a subcomponent?

Reply:
renderedChildren!